extract_endpoint/src/extract_endpoint/endpoint.py

```diff
@@ -0,0 +1,68 @@
+import os
+import base64
+import flask
+from werkzeug import utils
+from extract_endpoint import azure_utils
+from extract_endpoint import crypt_utils
+
+
+DEFAULT_ENDPOINT_SECRET_KEY = 'no key'
+
+
+App = flask.Flask(__name__)
+App.config.update(dict(
+    SECRET_KEY=os.environ.get('ENDPOINT_SECRET_KEY', DEFAULT_ENDPOINT_SECRET_KEY),
+    AZURE_COORDINATES=azure_utils.StorageCoordinates(
+        account=os.environ.get(azure_utils.EnvVariables.account.value, azure_utils.DefaultVariables.account.value),
+        key=os.environ.get(azure_utils.EnvVariables.key.value, azure_utils.DefaultVariables.key.value),
+        container=os.environ.get(azure_utils.EnvVariables.container.value,
+                                 azure_utils.DefaultVariables.container.value),
+        domain=os.environ.get(azure_utils.EnvVariables.domain.value, azure_utils.DefaultVariables.domain.value)
+    )
+))
+
+
+@App.route('/', methods=['GET'])
+def frontend() -> str:
+    return ''
+
+
+@App.route('/test_alive', methods=['GET'])
+def test_alive() -> str:
+    return 'Alive!'
+
+
+@App.route('/robots933456.txt', methods=['GET'])
+def robots() -> None:
+    flask.abort(404)
+
+
+@App.route('/upload_file', methods=['POST'])
+def upload_file() -> str:
+    if App.config['SECRET_KEY'] == DEFAULT_ENDPOINT_SECRET_KEY:
+        raise ValueError("Need to define environment variable ENDPOINT_SECRET_KEY")
+    if 'signature' not in flask.request.form:
+        flask.abort(404)
+    if 'salt' not in flask.request.form:
+        flask.abort(404)
+    if flask.request.form.get('filename', '') == '':
+        flask.abort(404)
+    if 'file' not in flask.request.files:
+        flask.abort(404)
```

Comment:
I feel like these should maybe be `400` errors, not `404`. It's not that there wasn't anything here, it's that the request is malformed/missing data

Reply:
Also, is there not some flask enum or something for aborting by a name? 
Even though these are pretty well known, it feels a little "magic number"-y. Maybe a variable `BAD_REQUEST` and `NOT_FOUND` that are `400` and `404` respectively?

Reply:
👍 to better status codes. (there doesn't seem to be any constants for them defined in flask)

Reply:
If there are no flask variables for them, I think we should make them ourselves, just some variables that describe the status code

Reply:
ah ha, `http.HTTPStatus` seems to be useful here.

Reply:
Yes it does 👍